src/engine/tests/Test_AbsoluteValueConstraint.h

```diff
@@ -296,1071 +296,470 @@ class AbsoluteValueConstraintTestSuite : public CxxTest::TestSuite
-            // 3 < x_b < 6 , 3 < x_f < 7
-            entailedTightenings.clear();
-            abs.getEntailedTightenings( entailedTightenings );
-            assert_lower_upper_bound( f, b, 3, 3, 6, 7, entailedTightenings );
-
-            abs.notifyUpperBound( f, 6 );
-            abs.notifyUpperBound( b, 7 );
-            // 3 < x_b < 6 , 3 < x_f < 6
-            //  --> x_b < 6
-            entailedTightenings.clear();
-            abs.getEntailedTightenings( entailedTightenings );
-            assert_lower_upper_bound( f, b, 3, 3, 6, 6, entailedTightenings );
-
-            abs.notifyLowerBound( f, -3 );
-            // 3 < x_b < 6 , 3 < x_f < 6
-            // --> 3 < x_f
-            entailedTightenings.clear();
-            abs.getEntailedTightenings( entailedTightenings );
-            assert_lower_upper_bound( f, b, 3, 3, 6, 6, entailedTightenings );
-
-            abs.notifyLowerBound( b, 5 );
-            abs.notifyUpperBound( f, 5 );
-            // 5 < x_b < 6 , 3 < x_f < 5
-            entailedTightenings.clear();
-            abs.getEntailedTightenings( entailedTightenings );
-            assert_lower_upper_bound( f, b, 5, 3, 6, 5, entailedTightenings );
-        }
+        abs.notifyLowerBound( b, 1 );
+        abs.notifyLowerBound( f, 2 );
+        abs.notifyUpperBound( b, 7 );
+        abs.notifyUpperBound( f, 7 );
+
+        // 1 < x_b < 7 , 2 < x_f < 7 -> 2 < x_b
+        abs.getEntailedTightenings( entailedTightenings );
+        assert_lower_upper_bound( f, b, 1, 2, 7, 7, entailedTightenings );
+
+        abs.notifyLowerBound( b, 3 );
+        // 3 < x_b < 7 , 2 < x_f < 7
+        entailedTightenings.clear();
+        abs.getEntailedTightenings( entailedTightenings );
+        assert_lower_upper_bound( f, b, 3, 2, 7, 7, entailedTightenings );
+
+        abs.notifyLowerBound( f, 3 );
+        abs.notifyUpperBound( b, 6 );
+        // 3 < x_b < 6 , 3 < x_f < 7
+        entailedTightenings.clear();
+        abs.getEntailedTightenings( entailedTightenings );
+        assert_lower_upper_bound( f, b, 3, 3, 6, 7, entailedTightenings );
+
+        abs.notifyUpperBound( f, 6 );
+        abs.notifyUpperBound( b, 7 );
+        // 3 < x_b < 6 , 3 < x_f < 6
+        //  --> x_b < 6
+        entailedTightenings.clear();
+        abs.getEntailedTightenings( entailedTightenings );
+        assert_lower_upper_bound( f, b, 3, 3, 6, 6, entailedTightenings );
+
+        abs.notifyLowerBound( f, -3 );
+        // 3 < x_b < 6 , 3 < x_f < 6
+        // --> 3 < x_f
+        entailedTightenings.clear();
+        abs.getEntailedTightenings( entailedTightenings );
+        assert_lower_upper_bound( f, b, 3, 3, 6, 6, entailedTightenings );
+
+        abs.notifyLowerBound( b, 5 );
+        abs.notifyUpperBound( f, 5 );
+        // 5 < x_b < 6 , 3 < x_f < 5
+        entailedTightenings.clear();
+        abs.getEntailedTightenings( entailedTightenings );
+        assert_lower_upper_bound( f, b, 5, 3, 6, 5, entailedTightenings );
```

Comment:
The TS_ASSERT macro needs to be invoked around the assert_lower_upper_bound() function. Otherwise cxxTest will report the assertion in the function definition, rather than in the test in which assertion is broken.